langwatch/src/components/AnnotationDrawer.tsx

```diff
@@ -72,22 +88,34 @@ export function AnnotationDrawer({
     defaultValues: {
       isThumbsUp: "thumbsUp",
       comment: comment,
+      scoreOptions: getAnnotation.data?.scoreOptions,
     },
   });
 
   const thumbsUpValue = watch("isThumbsUp");
 
   useEffect(() => {
-    if (getAnnotation.data) {
+    if (action === "edit") {
       const thumbValue = isThumbsUp === true ? "thumbsUp" : "thumbsDown";
       setValue("isThumbsUp", thumbValue);
       setValue("comment", comment);
+
+      Object.entries(scoreOptions ?? {}).forEach(([key, value]) => {
+        console.log("value", value);
+        unstable_batchedUpdates(() => {
```

Comment:
why do we need this?

Reply:
> if you're using React 18 and above, you do not need it anymore because React 18 now support automatic batching.
>
> This means that updates inside of timeouts, promises, native event handlers or any other event will batch the same way as updates inside of React events
> 
> you can read more about it [here](https://dev.to/jackbuchananconroy/react-18-what-s-changed-automatic-batching-13ec)

did you face any issues that you had to use this?